apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -871,77 +794,55 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     :one
   end
 
-  def fetch_datastream_maximum_storage_retention(client) do
-    maximum_storage_retention_statement = """
-    SELECT blobAsInt(value)
-    FROM kv_store
-    WHERE group='realm_config' AND key='datastream_maximum_storage_retention'
-    """
+  def fetch_datastream_maximum_storage_retention(realm) do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
 
     query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(maximum_storage_retention_statement)
-      |> DatabaseQuery.consistency(:quorum)
+      KvStore
+      |> where(group: "realm_config", key: "datastream_maximum_storage_retention")
+      |> select([v], fragment("blobAsInt(?)", v.value))
+      |> put_query_prefix(keyspace_name)
```

Comment:
It might be handy to use some wrapper APIs on KvStore to automatically handle parsing binary values into specific types, as we're doing in other services.

https://github.com/astarte-platform/astarte/blob/1e8147fcc5463d8bfb5916d306eb0858933491c0/apps/astarte_pairing/lib/astarte_pairing/realms/kv_store.ex#L89

Reply:
After IRL discussion, we've decided to wait before going with this solution.